### ISSUE TITLE
Got HTTP ERROR 431 when Enable AAD Auth for Azure Web App

### ISSUE THREAD

**Issue by huyenvv90** [microsoft/spektate#167]
The first time, I've followed this [guideline](https://github.com/microsoft/spektate/blob/master/WebAppHosting.md#enable-aad-auth-for-azure-web-app) and bypass the Auth AAD step. It's worked.
But when I on the Auth AAD, It's not worked and got the issue (see image)
When I off the Auth Azure AD, It's worked again.
![image](https://user-images.githubusercontent.com/69791411/116984500-839cfa00-acf5-11eb-8e9c-35fb0792aad6.png)

**Comment by andrebriggs** [comment]
Hi @huyenvv90 sorry for the issues you're encountering. Seems like AAD Auth is causing issues. Is your use case being able to access Spektate publicly but with authorization?

@samiyaakhtar how difficult would it be to set up an identity provider in Spektate?

Given an AAD application client id and password we could authenticate. The set up could be as follows:

1. Configure Spektate with AAD Identity provider credentials for a tenant
2. Run Spektate in your Kubernetes cluster 
3. Setup an Azure public IP address 
4. Setup an ingress in your Kubernetes cluster using the public IP address (Your can configure such that only Azure Front Door as access)
5. Setup something like Azure Front Door to route to your backend (exposed public IP address) 

This way folks can access Spektate security with authorization using Kubernetes to run the workload.

**Comment by huyenvv90** [comment]
yes, I can access Spektate publicly. But when I setup AAD Auth, it's returned 431 error.

**Comment by samiyaakhtar** [comment]
@andrebriggs I think it's worth looking into, I will open an issue to track this
